packages/app-store/stripepayment/api/callback.ts

```diff
@@ -19,8 +20,13 @@ function getReturnToValueFromQueryState(req: NextApiRequest) {
 
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
   const { code, error, error_description } = req.query;
+  const state = decodeOAuthState(req);
 
   if (error) {
+    // User cancels flow
+    if (error === "access_denied") {
+      state?.onErrorReturnTo ? res.redirect(state.onErrorReturnTo) : res.redirect("/apps/installed/payment");
```

Comment:
Do we need to getSafeRedirectUrl here like we have the others? Not too familiar with this code so let me know if im wrong (and why so i learn a thing or two :P )

Reply:
I'd like to have this in place. There's no harm in the way we're handling things but this could be a good safe guard if a self-hoster wants to make changes.